receiver/windowseventlogreceiver/receiver_windows_test.go

```diff
@@ -188,91 +168,68 @@ func TestReadWindowsEventLoggerRaw(t *testing.T) {
-	}()
-
-	logger, err := eventlog.Open(src)
-	require.NoError(t, err)
-	defer logger.Close()
-
-	err = logger.Info(10, logMessage)
-	require.NoError(t, err)
-
-	logsReceived := func() bool {
-		return sink.LogRecordCount() == 0
+	excludedSrc := "otel-excluded-windowseventlogreceiver-test"
+	notExcludedSrc := "otel-windowseventlogreceiver-test"
+	testSources := []string{excludedSrc, notExcludedSrc}
+
+	for _, src := range testSources {
+		uninstallEventSource, err := assertEventSourceInstallation(t, src)
+		defer uninstallEventSource()
+		require.NoError(t, err)
 	}
 
-	// logs sometimes take a while to be written, so a substantial wait buffer is needed
-	require.Eventually(t, logsReceived, 10*time.Second, 200*time.Millisecond)
-	results := sink.AllLogs()
-	require.Len(t, results, 0)
-}
-
-func TestReadWindowsEventLoggerRawWithExcludeProvider(t *testing.T) {
-	logMessage := "Test log"
-	src := "otel"
-
-	ctx := context.Background()
-	factory := newFactoryAdapter()
-	createSettings := receivertest.NewNopSettings()
-	cfg := createTestConfig()
-	cfg.InputConfig.Raw = true
-	cfg.InputConfig.ExcludeProviders = []string{src}
-	sink := new(consumertest.LogsSink)
-
-	receiver, err := factory.CreateLogsReceiver(ctx, createSettings, cfg, sink)
-	require.NoError(t, err)
-
-	err = receiver.Start(ctx, componenttest.NewNopHost())
-	require.NoError(t, err)
-	defer func() {
-		require.NoError(t, receiver.Shutdown(ctx))
-	}()
-
-	err = eventlog.InstallAsEventCreate(src, eventlog.Info|eventlog.Warning|eventlog.Error)
-	defer func() {
-		require.NoError(t, eventlog.Remove(src))
-	}()
-	require.NoError(t, err)
-
-	logger, err := eventlog.Open(src)
-	require.NoError(t, err)
-	defer logger.Close()
-
-	err = logger.Info(10, logMessage)
-	require.NoError(t, err)
-
-	logsReceived := func() bool {
-		return sink.LogRecordCount() == 0
+	tests := []struct {
+		name string
+		raw  bool
+	}{
+		{
+			name: "with_EventXML",
+		},
+		{
+			name: "with_Raw",
+			raw:  true,
+		},
 	}
 
-	// logs sometimes take a while to be written, so a substantial wait buffer is needed
-	require.Eventually(t, logsReceived, 10*time.Second, 200*time.Millisecond)
-	results := sink.AllLogs()
-	require.Len(t, results, 0)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			factory := newFactoryAdapter()
+			createSettings := receivertest.NewNopSettings()
+			cfg := createTestConfig()
+			cfg.InputConfig.Raw = tt.raw
+			cfg.InputConfig.ExcludeProviders = []string{excludedSrc}
+			sink := new(consumertest.LogsSink)
+
+			receiver, err := factory.CreateLogsReceiver(ctx, createSettings, cfg, sink)
+			require.NoError(t, err)
+
+			err = receiver.Start(ctx, componenttest.NewNopHost())
+			require.NoError(t, err)
+			defer func() {
+				require.NoError(t, receiver.Shutdown(ctx))
+			}()
+			// Start launches nested goroutines, give them a chance to run before logging the test event(s).
+			time.Sleep(3 * time.Second)
```

Comment:
maybe consider a assert.EventuallyWithT to test this multiple times until it works

Reply:
In principle, I really dislike sleeps on tests. I tried a few alternative before settling on this, however, I couldn't make it work reliably without a sleep.